File: frontend/src/api/lost_found.ts
import request from '@/utils/request'
import type { LostFoundItem, LostFoundComment } from '@/types'

export function getLostFoundItems(params?: { type?: string; status?: string }) {
  return request.get<LostFoundItem[]>('/lost-found/items', { params })
}

export function getLostFoundItem(id: number) {
  return request.get<LostFoundItem>(`/lost-found/items/${id}`)
}

export function createLostFoundItem(data: {
  type: string
  title: string
  description?: string
  location?: string
  contact?: string
  image_url?: string
}) {
  return request.post<LostFoundItem>('/lost-found/items', data)
}

export function updateLostFoundStatus(id: number, status: string) {
  return request.put<LostFoundItem>(`/lost-found/items/${id}/status`, { status })
}

export function deleteLostFoundItem(id: number) {
  return request.delete(`/lost-found/items/${id}`)
}

export function createLostFoundComment(id: number, content: string) {
  return request.post<LostFoundComment>(`/lost-found/items/${id}/comments`, { content })
}

File: backend/app/models/lost_found.py
from datetime import datetime, timezone
from sqlalchemy import String, Text, ForeignKey, DateTime
from sqlalchemy.orm import Mapped, mapped_column
import enum

from app.core.database import Base


class ItemType(str, enum.Enum):
    LOST = "lost"
    FOUND = "found"


class ItemStatus(str, enum.Enum):
    OPEN = "open"
    CLAIMED = "claimed"
    CLOSED = "closed"


class LostFoundItem(Base):
    __tablename__ = "lost_found_items"

    id: Mapped[int] = mapped_column(primary_key=True)
    user_id: Mapped[int] = mapped_column(ForeignKey("users.id"), index=True)
    type: Mapped[ItemType] = mapped_column(Text)
    title: Mapped[str] = mapped_column(String(100))
    description: Mapped[str] = mapped_column(Text, default="")
    location: Mapped[str] = mapped_column(String(200), default="")
    contact: Mapped[str] = mapped_column(String(200), default="")
    image_url: Mapped[str | None] = mapped_column(String(500))
    status: Mapped[ItemStatus] = mapped_column(Text, default=ItemStatus.OPEN)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), default=lambda: datetime.now(timezone.utc))


class LostFoundComment(Base):
    __tablename__ = "lost_found_comments"

    id: Mapped[int] = mapped_column(primary_key=True)
    item_id: Mapped[int] = mapped_column(ForeignKey("lost_found_items.id"), index=True)
    user_id: Mapped[int] = mapped_column(ForeignKey("users.id"))
    content: Mapped[str] = mapped_column(Text)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), default=lambda: datetime.now(timezone.utc))
